src/cryptsetup/cryptsetup-generator.c

```diff
@@ -389,26 +389,35 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                         return log_oom();
 
         } else if (streq(key, "luks.key")) {
+                int n;
 
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
-                r = sscanf(value, "%m[0-9a-fA-F-]=%ms", &uuid, &uuid_value);
-                if (r == 2) {
+                r = sscanf(value, "%m[0-9a-fA-F-]=%n", &uuid, &n);
```

Comment:
Hmm, let's say that the `%m` was matched, so the return value is increased up to 1, but then then we don't match "=" and %n is not assigned. We cannot distinguish this case from successful conversion. So I think my suggestion to use `sscanf` was wrong. Sorry.

What about the following:
```C
int n = strspn(value, LETTERS DIGITS "-");
if (n > 0 && value[n] == '=') {
    keyspec = value + n + 1;
    ...
}
```
?

Reply:
It feels like that checking the return value of ```scanf()``` becomes pointless. E.g. with the proposed change we would confuse the following case, ```luks.key=foobar=```. Thus, we also need to check length of the keyspec and fallback to code from ```else``` branch if it is 0.

Reply:
Ok, I take that back. Keyfile path must be an absolute path so the example I gave is not valid.